### PR TITLE
8262460: Create release notes for JavaFX 16

### PR DESCRIPTION
**Pull request by abhinayagarwal**
Add release notes for JavaFX 16 to the repository

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8262460](https://bugs.openjdk.java.net/browse/JDK-8262460): Create release notes for JavaFX 16

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Johan Vos](https://openjdk.java.net/census#jvos) (@johanvos - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jfx pull/414/head:pull/414`
`$ git checkout pull/414`

### CODE REVIEW

**Comment by kevinrushforth** [comment]
The target branch should be changed to `jfx16`

**Comment by kevinrushforth** [comment]
And this means you will need to rebase on top of `jfx16`, so that it doesn't bring in commits from master.

File: doc-files/release-notes-16.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+Issue key|Summary|Subcomponent
+---------|-------|------------
+[JDK-8253597](https://bugs.openjdk.java.net/browse/JDK-8253597)|TreeTableView: must select leaf row on click into indentation region|controls
+[JDK-8253696](https://bugs.openjdk.java.net/browse/JDK-8253696)|WebEngine refuses to load local "file:///" CSS stylesheets when using JDK 15|web
+[JDK-8248490](https://bugs.openjdk.java.net/browse/JDK-8248490)|[macOS] Undecorated stage does not minimize|window-toolkit
+[JDK-8252446](https://bugs.openjdk.java.net/browse/JDK-8252446)|Screen.getScreens() is empty sometimes|graphics
+[JDK-8251941](https://bugs.openjdk.java.net/browse/JDK-8251941)|ListCell: visual artifact when items contain null values|controls
+[JDK-8176270](https://bugs.openjdk.java.net/browse/JDK-8176270)|Adding ChangeListener to TextField.selectedTextProperty causes StringOutOfBoundsException|controls
+[JDK-8255415](https://bugs.openjdk.java.net/browse/JDK-8255415)|Nested calls to snap methods in Region give different results|graphics
+[JDK-8246343](https://bugs.openjdk.java.net/browse/JDK-8246343)|Fix mistakes in FX API docs|other
```
Comment (nlisker) [inline doc-files/release-notes-16.md:20]: This one is from an older version.

**Review by nlisker — CHANGES_REQUESTED** [review]

File: doc-files/release-notes-16.md (hunk 1 above)
Comment (abhinayagarwal) [reply doc-files/release-notes-16.md:20]: All original issues are added to the Release Notes if an issue is a backport issue. It seems like there is backport issue against JDK-8246343 with fix version `openjfx16`.

Comment (johanvos) [reply doc-files/release-notes-16.md:20]: There are some "backport" issues that are sort of forward ported but still have type backport. For example, issues fixed in 15.0.1 are "backported" to 16.

Comment (kevinrushforth) [reply doc-files/release-notes-16.md:20]: There are issues in 15 with a backport to 16. I'd be surprised if that were the case for 15.0.1 (and if so, they would still be in the release notes for 16 if they didn't ship with 15 GA).

https://bugs.openjdk.java.net/browse/JDK-8246343 is the case of a bug that was fixed in 15 and then synced into mainline for 16. So it should be ignored.

Comment (johanvos) [reply doc-files/release-notes-16.md:20]: Right, I was confused when I saw the tag 15.0.1+0 but it is indeed in 15.

**Comment by kevinrushforth** [comment]
A couple other comments:

1. Previous release notes were sorted first by subcomponent and then by bug ID. You might consider whether to do that again for 16?
2. The following bug was called out during the CSR as needing a release note to highlight the new log warning:
[JDK-8256362](https://bugs.openjdk.java.net/browse/JDK-8256362): JavaFX must warn when the javafx.* modules are loaded from the classpath
I think I'm on the hook for that one, so I can provide you the proposed text later today, in case you want to include it.

File: doc-files/release-notes-16.md (hunk 1 above)
Comment (nlisker) [reply doc-files/release-notes-16.md]: There might be other issues like that. Maybe the backport issue type should be filtered.

Comment (abhinayagarwal) [reply doc-files/release-notes-16.md]: This has already been taken care of. I have removed all the issues which were already shipped with 15GA and then backported to 16.

**Comment by kevinrushforth** [comment]
For the last couple releases, I generated the list of bugs to add to release notes, and yes, I typically exclude backports for that reason (it isn't perfect, but it also excludes the issues labeled with the `hgupdate-sync` label, which can be added to indicate an issue synced in from a prior release).

Here is the same filter I used in the past, updated for JavaFX 16 for comparison:

https://bugs.openjdk.java.net/issues/?filter=40381

You'll notice that it excludes test bugs and build issue (so the "GitHub Actions" bugs don't show up, for example)

**Comment by kevinrushforth** [comment]
Below is the release note for the warning, taken from [this release note sub-task](https://bugs.openjdk.java.net/browse/JDK-8262948):

---

### JavaFX runtime logs a warning if javafx.* modules are loaded from the classpath

The JavaFX classes must be loaded from a set of named `javafx.*` modules on the _module path_. Loading the JavaFX classes from the classpath is not supported. The JavaFX runtime logs a warning at startup if the JavaFX classes are not loaded from the expected named module. See [JDK-8256362](https://bugs.openjdk.java.net/browse/JDK-8256362) for more information.

**Comment by kevinrushforth** [comment]
I also missed noticing that the following bug is marked with the need for a release note:

[JDK-8196079](https://bugs.openjdk.java.net/browse/JDK-8196079): Remove obsolete Pisces rasterizer

I'll create one shortly and add a comment to this PR when done.

**Comment by kevinrushforth** [comment]
Below is the release note for the removal of the Pisces rasterizer, taken from [this release note sub-task](https://bugs.openjdk.java.net/browse/JDK-8262951):

---

### The obsolete Pisces rasterizer has been removed from JavaFX

The obsolete Pisces rasterizer has been removed from JavaFX. The Marlin rasterizer has been the default since JDK 10, but it was possible to select either the native Pisces rasterizer or the Java-based Pisces rasterizer by setting the `prism.rasterizerorder` system property to `nativepisces` or `javapisces`, respectively. Those options will now be silently ignored and the default Marlin rasterizer will always be used. See [JDK-8196079](https://bugs.openjdk.java.net/browse/JDK-8196079) for more information.

**Comment by abhinayagarwal** [comment]
> https://bugs.openjdk.java.net/issues/?filter=40381

Hi @kevinrushforth ,

Thanks for this list. You may want to add `test_sprint` to the list of labels :)

**Comment by kevinrushforth** [comment]
I don't think there should be any `test_sprint` labeled bugs that don't also have one of `testbug` or `noreg-self`, but I'll check.

**Comment by kevinrushforth** [comment]
And it turns out there were three. I added the `testbug` label. Thanks for pointing it out.

**Review by kevinrushforth — APPROVED** [review]
Looks good.

**Review by johanvos — APPROVED** [review]

**Pull request closed without merging**